extensions/VisualEditor/wikia/modules/ve/ui/pages/ve.ui.WikiaParameterPage.js

```diff
@@ -4,31 +4,156 @@
+		}
+
+		win.bookletLayout.$element.html( '' );
+		win.initializeTemplateParameters();
+	}
+
+	/**
+	 * shorthand function to open dialogs
+	 *
+	 * @param {string} name - Name of a dialog to open
+	 */
+
+	function openDialog( name ) {
+		ve.ui.actionFactory
+			.create( 'window', ve.init.target.getSurface() )
+			.open( name );
+	}
+	/**
+	 *
+	 * @param {ve.ui} currentWindow
+	 * @param {ve.dm.WikiaCart} cartItems
+	 * @returns {string|null}
+	 */
+	function getImageName( currentWindow, cartItems ) {
+		if (
+			currentWindow.currentAction &&
+			currentWindow.currentAction.getAction() === 'goback' &&
+			cartItems.length > 0
+		) {
+			return cartItems[0].data;
+		}
+
+		return null;
+	}
+	/**
+	 * In order to intercept default behaviour of wikiaMediaInsert
+	 * additional action was added to the dialog
+	 * goback - it does nothing, but allows us to get images chosen in the dialog
+	 *
+	 * @param {string} action action name (e.g. goback, apply)
+	 */
+	function setDefaultMediaInsertDialogAction( action ) {
+		ve.ui.WikiaMediaInsertDialog.static.actions[0].action = action;
+	}
+
+	/**
+	 * Event handler on image button
+	 * Opens wikiaMediaInsert dialog and allows user to choose an image
+	 * from wiki images
+	 *
+	 * @param {string} paramName Name of a field (e.g. image1)
+	 */
+	function handleClickOnImageButton( paramName ) {
+		var windowManager = ve.init.target.getSurface().getDialogs(),
+			transclusionModel = windowManager.currentWindow.transclusionModel,
+			windowName = windowManager.currentWindow.constructor.static.name;
+
+		windowManager.closeWindow( windowName ).done( function () {
+			openDialog( 'wikiaMediaInsert' );
+			setDefaultMediaInsertDialogAction( 'goback' );
+
+			windowManager.once( 'closing', function ( currentWindow ) {
+				var imageName;
+
+				if ( currentWindow instanceof ve.ui.WikiaMediaInsertDialog ) {
+					imageName = getImageName( currentWindow, currentWindow.cart.getItems() );
+
+					windowManager.closing.done( function () {
+						/**
+						 * first 'opening' is an event fired when a dialog is opening
+						 * in the callback to this event I get a windowManager in a state where
+						 * windowManager.opening is actually a promise
+						 * before that windowManager is null
+						 */
+						windowManager.once( 'opening', function ( win ) {
+							windowManager.opening.done(
+								restoreInfoboxDialog.bind( this, win, transclusionModel, imageName, paramName )
+							);
+						});
+
+						setDefaultMediaInsertDialogAction( 'apply' );
+						openDialog( windowName );
+					});
+				}
+			});
+		});
+	}
+
+	/**
+	 * Wikia transclusion dialog template page.
+	 *
+	 * @class
+	 * @extends ve.ui.MWParameterPage
+	 *
+	 * @constructor
+	 * @param {ve.dm.MWParameterModel} parameter Template parameter
+	 * @param {string} name Unique symbolic name of page
+	 * @param {Object} [config] Configuration options
+	 */
+	ve.ui.WikiaParameterPage = function VeUiWikiaParameterPage( parameter, name, config ) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Make this function anonymous by removing its name: 'function() {...}'. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3ANamedFunctionExpression)